server/src/mindustry/server/ServerControl.java

```diff
@@ -172,6 +172,18 @@ public ServerControl(String[] args){
             }
         });
 
+        Events.on(WaveEvent.class, e -> {
+            if(state.is(State.playing) && Config.valueOf("enableAutosave").bool() ){
+                Fi file = saveDirectory.child("autosave-" + Strings.capitalize(world.getMap().name()) + "-" + state.wave + "." + saveExtension);
+
+                Core.app.post(() -> {
+                    SaveIO.save(file);
```

Comment:
This should be `try-catch`-ed and `Log.err` the exception, if any occurs. An autosave failure due to lack of disk space or an IO error shouldn't crash the server.